bugbug/model.py

```diff
@@ -567,7 +569,19 @@ def train(self, importance_cutoff=0.15, limit=None):
 
             self.clf.fit(X_train, self.le.transform(y_train))
 
-        with open(self.__class__.__name__.lower(), "wb") as f:
+        model_directory = self.__class__.__name__.lower()
+        makedirs(model_directory, exist_ok=True)
+
+        if issubclass(type(self.clf), XGBModel):
+            xgboost_model_path = path.join(model_directory, "xgboost.ubj")
+            self.clf.save_model(xgboost_model_path)
+
+            # Since we save the classifier separately, we need to clear the clf
+            # attribute to prevent it from being pickled with the model object.
+            self.clf = self.clf.__class__(n_jobs=self.clf.n_jobs)
```

Comment:
Why are we passing n_jobs here? What if a submodel uses other parameters than n_jobs?

Reply:
This would be valid in older versions of xgboost because it was the only parameter that would be used before copying everything from the JSON dump. However, I just noticed that this got changed in https://github.com/dmlc/xgboost/pull/8963 where none of the hyper-parameters will be saved/loaded. 

^^ This reminds us to never rely on implementation details :)

Currently, `n_jobs` is the only parameter that we use when we instantiate our xgboost models. We may file a follow-up issue to store hyper-parameters in an attribute (dictionary) so we use it when instantiating a model for `clf`.

Reply:
I wonder why it isn't just saved along with the model!

Yeah, we should file a follow-up for this.

Reply:
> I wonder why it isn't just saved along with the model!

It was doing that, but that got changed: "The scikit-learn interface `save_model` will save only the model and discard all hyper-parameters. This is to align with the native XGBoost interface, which distinguishes the hyper-parameter with model parameters."[^1]

[^1]: https://github.com/dmlc/xgboost/pull/8963

> Yeah, we should file a follow-up for this.

I filed an issue: https://github.com/mozilla/bugbug/issues/3794